### DISCUSSION TITLE
Support for Pydantic

### DISCUSSION THREAD

**Discussion by hardbyte** [hardbyte/ha-evnex#35] [General]
Because homeassistant shares a Python environment dependencies have to be compatible across the ecosystem.

There is a push to update to Pydantic V2, and it appears enough components/integrations have updated in the 2024.5 release that I think it is now safe to assume pydantic 2+ for versions 0.6 onwards. This means if you have older integrations or custom components that depend on Pydantic v1 you will not be able to install evnex.

If anyone wants to try keep compatibility with both pydantic v1 and v2 I'm happy to review pull requests for a v0.5.X release. But `main` is going to move on.

**Comment by colinjgibbs** [comment]
Hi, I assume the version of pydantic you refer to is the one installed in the docker container "homeassistant".  I ask because I upgraded that to version 2 yesterday and still Evnex 0.6 fails the install as it tells me that the pydantic version is still 1.x. here is the out put from pip:
_homeassistant:/# pip show pydantic
Name: pydantic
Version: 2.10.1
Summary: Data validation using Python type hints
Home-page: https://github.com/pydantic/pydantic_

Or do I need to upgrade pydantic somewhere else?  I have currently gone back to 0.53 of the Evnex integration.

**Reply by hardbyte** [reply]
Yeah not a bad idea to stick with v0.5.X until home assistant itself is released with pydantic v2+.

I'm running `home-assistant:2024.11` in docker. If I exec into the container:

```
# pip show pydantic                                          
Name: pydantic                                                                   
Version: 2.9.2
# which pip                                                  
/usr/local/bin/pip                                                               
# pip freeze | grep pydantic                                 
pydantic==2.9.2                                                                  
pydantic_core==2.23.4
```

**Reply by colinjgibbs** [reply]
Yup I'll stick with 0.5.3 for now.

**Reply by colinjgibbs** [reply]
Just confirming that since HA went to pydantic 2.x with the 2025 release version 0.6 works fine.  Thanks!
